core/chains/evm/types/nonce.go

```diff
@@ -0,0 +1,14 @@
+package types
+
+import "fmt"
+
+var _ fmt.Stringer = &Nonce{}
+
+// Nonce wraps an EVM nonce into a stringable type
+type Nonce struct {
+	N int64
+}
```

Comment:
Is this possible? It would be more efficient:
```suggestion
type Nonce int64
```

Reply:
I tried it initially. But doesn't work.
Any caller that expects an int64 nonce, cannot use Nonce.

Reply:
Can you please elaborate? If you define `type Nonce int64`, then any `var n Nonce` can be trivially converted directly `int64(n)`.

Reply:
<img width="1116" alt="image" src="https://user-images.githubusercontent.com/34992934/232943250-0010c667-4093-4c32-9556-57481a51a54b.png">

Reply:
You can convert it directly, in place or via a helper method:
![image](https://user-images.githubusercontent.com/1194128/232948281-8996f5af-2710-42bc-903e-83d3745aad12.png)

Reply:
Tagging @aalu1418 too for another comment on this same thing.

But is that option better than current?

Also, the helper methods don't work generically.


The below code still fails:
```

type Nonce int64

func (n Nonce) Int64() int64 {return int64(n)}

func DoSomething(n *int64) {
	// blah
}

func main() {
	var n Nonce = 12
	strconv.FormatInt(n.Int64(), 10)

	DoSomething(&n.Int64())
}


```

Reply:
afaik you usually cannot create a new "object" and then take the reference of it immediately - that's just golang

also similar question to the address + hash types, why are we passing the nonce as a reference?

Reply:
I know you cannot take ref of a temporary object. So we have to do this:
```
nVal := n.Int64()
DoSomething(&nVal)
```

Instead, in the other solution, you just need single line:
`DoSomething(&n.N)`

Isn't the second option more concise?


In any way, I don't have any strong opinions here, and feel this is all just nit. I don't understand why it is more efficient to use "type Nonce int64".

Reply:
It is more concise, but it is meaningfully different to pass a pointer to internals of another type - do we actually need to do that for some kind of mutation case? If not, using a pointer doesn't generally make sense for primitive types that can be passed by value, using the same or less memory than the pointer itself.

Reply:
Yes, we do need a ref to nonce here:
https://github.com/smartcontractkit/chainlink/blob/4c63c2a3f1cff445ab0d692e4fefa9f6225fdeb8/core/chains/evm/txmgr/models.go#L186

In any case, I don't have strong opinions, so switched to what you're suggesting. But at above place, it adds more conversion steps, and also, have to be careful while using it in SQL queries.
The DB SQL calls accept interface{}, but if the type doesn't match what they expect, they fail in runtime. So easy to make mistake passing Nonce where int64 was expected.

Reply:
Given:
```go
type Nonce int64
```
Since you can already do this directly:
```go
var n Nonce
_ = int64(n)
```
You could define the method to return a pointer:
```go
func (n Nonce) Int64() *int {
	return (*int)(&n)
}
```
No extra lines, and this has the benefit of returning a pointer to a newly allocated variable, rather than the one inside the original Nonce.

Reply:
Ok, nice, yes, that works.
So there are many places which require int64 and not *int64. So if I change the above to *int64, I will have to change all other places too.

Also thinking more about above Nonce *int64 type, that struct(EthTx) is also getting generalized in a future PL, into a SEQ type. So then we no longer will need a reference to Nonce.

Thus keeping it as it is for now, is good.